### PR TITLE
Add proxy pass through test

### PR DESCRIPTION
**Pull request by nonhermitian**
### Summary
Test is bad proxy info raises proxyerror in the API.  This means we correctly pass the `proxies` dict to the API.  Note that this test takes a while because the API does not check if a connection can be established before trying the max number of times.

### CODE REVIEW

File: test/python/ibmq/test_registration.py [hunk 1]
```diff
@@ -143,6 +143,20 @@ def test_delete_all_accounts(self):
             self.assertEqual(len(qiskit.IBMQ._accounts), 0)
             self.assertEqual(len(read_credentials_from_qiskitrc()), 0)
 
+    @requires_qe_access
+    def test_pass_bad_proxy(self, qe_token, qe_url):
+        """Test proxy pass through."""
+        prox = {'urls': {
+            'http': 'http://user:password@1.2.3.4:5678',
+            'https': 'https://user:password@1.2.3.4:5678'}}
```
Comment (diego-plan9) [inline test/python/ibmq/test_registration.py]: Can you replace the host with either an invalid one, or one that resolves quickly: perhaps `127.0.0.1`? This way hopefully we avoid the delay, while still getting the error.

Comment (nonhermitian) [reply test/python/ibmq/test_registration.py]: ok

**Comment by nonhermitian** [comment]
The issue is really not the url that is being connected to, but rather the requests module does not care if it cannot establish a connection, it keeps trying up until the max number of connection attempts is made.

**Comment by diego-plan9** [comment]
> The issue is really not the url that is being connected to, but rather the requests module does not care if it cannot establish a connection, it keeps trying up until the max number of connection attempts is made. See issue #992

Hmm sounds odd - running the tests locally with `127.0.0.1` results in it passing in less than a second, which is actually sort of consistent with your explanation: it attempts to open a connection to _the proxy_ a number of times (which instantly fail), and gives up after `max_retries` attempts. Do you have a proxy yourself on that port? 

Anyway, seems part of another issue indeed - the PR itself is looking ready!

**Pull request merged**